js/api/Surveys.js

```diff
@@ -139,7 +159,8 @@ function clearSurveyCache(exclusions) {
 
     // Standard JS array.filter doesn't work with these Realm objects, so we have to take care of this filtering manually.
     // Current version of Realm.io does not support exclusion queries for strings.
-    for (var i = surveys.length - 1; i >= 0; i--) {
+    const surveysLength = surveys.length;
+    for (var i = surveysLength - 1; i >= 0; i--) {
```

Comment:
There are other cases in which these could be replace across the code as well. I'll take care of the ones in this PR, the other I'll do under another story to avoid too many conflicts.